### PR TITLE
Add chalk and do some other stuff

### PR DESCRIPTION
**Pull request by rogerhutchings**
- Adds chalk and colorises the output
- Add a formatter (prettier is awesome)
- Update package.json with some useful fings

Edit: add screenshot

Screenshot 2021-04-12 at 08 30 57

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -16,31 +17,40 @@ const getHP = async (browser, player) => {
...
+    (el) => el.innerText
+  )
   const half = Math.floor(parseInt(max) / 2)
 
-  return {...player, current, max, half}
+  return { ...player, current, max, half }
 }
 
-(async () => {
+;(async () => {
```
Comment (ghickman) [inline script.js:39]: What's the semi-colon doing in the syntax here?

**Review by ghickman — COMMENTED** [review]
Can you pop a screenshot in the PR body so I can see what the colourised output looks like?

File: script.js (hunk 1 above)
Comment (rogerhutchings) [reply script.js:39]: This gets added by prettier to stop the IIFE being executed as an argument to any preceding function:

```js
mybrokenfunction = function(){

} //no semicolon here
(function(g){


})(this);
```

actually becomes

```js
mybrokenfunction = function(){}(function(g){})(this);
```

Automatic Semicolon Insertion (ASI) is a Whole Thing

(source: https://stackoverflow.com/a/17978926)

Comment (ghickman) [reply script.js:39]: Ah!

And we need to make a function so we can mark it async AIUI?  The alternative would be `const foo = async () …` and the invoking it?

Comment (rogerhutchings) [reply script.js:39]: exactly

Comment (rogerhutchings) [reply script.js:39]: also actually true even if it wasn't async, an IIFE is just a way of immediately running a thing without a separate invocation:

```js
(function() {
    console.log('foo')
})()

// vs

function logFoo() {
    console.log('foo')
}
logFoo()
```

**Pull request merged**